File: backend/src/pre_autorizacion/features/policies/domain/entities/policy.py
"""Policy — póliza de seguro de un paciente.
PRD §4.2.2 (DB Notion 4: Pólizas).

Port directo de `frontend/src/app/features/policies/domain/entities/policy.ts`.

Decisiones de mapping TS → Python:
- `startDate` / `endDate` (string ISO en TS) → `datetime.date` en el dominio.
  El parseo string → date vive en infrastructure (DTOs / repos).
- `status` (literal union TS) → `PolicyStatus` StrEnum con los mismos 3 valores
  canónicos en inglés. El mapeo desde el prototipo (`'ACTIVA'`, etc.) sucede en seed.
"""

from __future__ import annotations

from dataclasses import dataclass
from datetime import date
from enum import StrEnum


class PolicyStatus(StrEnum):
    """PolicyStatus — estado canónico de una póliza."""

    ACTIVE = "ACTIVE"
    INACTIVE = "INACTIVE"
    SUSPENDED = "SUSPENDED"


@dataclass(frozen=True, slots=True)
class Policy:
    """Policy — entidad inmutable de póliza."""

    number: str
    patient_id: str
    plan: str
    insurer_id: str
    start_date: date
    end_date: date
    status: PolicyStatus

File: frontend/src/app/features/policies/domain/entities/policy.ts
/**
 * Policy — póliza de seguro de un paciente.
 * PRD §4.2.2 (DB Notion 4: Pólizas).
 *
 * NOTA: el prototipo data.js usa `status: 'ACTIVA'` (string en español).
 * Acá normalizamos a literales canónicos en inglés ('ACTIVE' | 'INACTIVE' | 'SUSPENDED')
 * y el seed se encarga del mapeo. Esto facilita lógica determinística posterior.
 *
 * `insurerId` referencia el `Insurer.id`. El seed lo deriva del campo `insurer` (string)
 * del prototipo (única aseguradora: "Aseguradora Andina S.A." → INS-ANDINA).
 */
export interface Policy {
  readonly number: string;
  readonly patientId: string;
  readonly plan: string;
  readonly insurerId: string;
  readonly startDate: string;
  readonly endDate: string;
  readonly status: 'ACTIVE' | 'INACTIVE' | 'SUSPENDED';
}
